pvlib/tests/test_windspeed.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import pandas as pd
+import pytest
+from pvlib import windspeed
+
+from .conftest import assert_series_equal
+from numpy.testing import assert_allclose
+
+
+@pytest.mark.parametrize(
+    'wind_speed_measured,height_measured,height_desired,wind_speed_calc',
+    [
+        (10, -2, 5, np.nan),
+        (-10, 2, 5, np.nan),
+        (5, 4, 5, 5.067393209486324),
+        (7, 6, 10, 7.2178684911195905),
+        (10, 8, 20, 10.565167835216586),
+        (12, 10, 30, 12.817653329393977)])
+def test_wind_speed_at_height_hellmann(wind_speed_measured,
+                                       height_measured,
+                                       height_desired,
+                                       wind_speed_calc):
+    result = windspeed.wind_speed_at_height_hellmann(
+        wind_speed_measured,
+        height_measured,
+        height_desired,
+        surface_type='unstable_air_above_open_water_surface')
+    assert_allclose(result, wind_speed_calc)
+
+
+@pytest.fixture
+def times():
+    return pd.date_range(start="2015-01-01 00:00", end="2015-01-01 05:00",
+                         freq="1h")
+
+
+@pytest.fixture
+def wind_speeds_measured(times):
+    return pd.Series([10, -10, 5, 7, 10, 12], index=times)
+
+
+@pytest.fixture
+def heights_measured(times):
+    return np.array([-2, 2, 4, 6, 8, 10])
+
+
+@pytest.fixture
+def heights_desired():
+    return np.array([5, 5, 5, 10, 20, 30])
+
+
+@pytest.fixture
+def wind_speeds_calc(times):
+    return pd.Series([np.nan, np.nan, 5.067393209486324, 7.2178684911195905,
+                      10.565167835216586, 12.817653329393977], index=times)
```

Comment:
I don't want to nitpick too much the tests since that's where we can only taste freedom, but two points:

1. If you need to test scalar inputs, I think it's just better to take the zero or last indexed values of each premise/expected series. Either with the `.iloc` method on a series object, or with the usual `[0]`|`[-1]` on the numpy view. You know, you already are checking that the values are correct with respect to the expected behaviour with the numpy and series tests with `assert_allclose`, and that's perfectly fine. There's usually no need to duplicate data in python.
2. And lastly, I think I mislead you in the test patterns _that I prefer_ in some previous PR when I told you about some example fixture. I just took the simpler one I found for the sake of simplifying without looking too much at its role in the code. Generally, I would group all the fixtures into one:

```python
@pytest.fixture
def wind_speeds_at_heighth_hellman():
    data = pd.DataFrame(
        index=pd.date_range(start="2015-01-01 00:00", end="2015-01-01 05:00", freq="1h"),
        columns=["wind_ref", "height_ref", "height_desired", "wind_desired"],  # or wind_expected
        data=[
            (10, -2, 5, np.nan),
            (-10, 2, 5, np.nan),
            (5, 4, 5, 5.067393209486324),
            (7, 6, 10, 7.2178684911195905),
            (10, 8, 20, 10.565167835216586),
            (12, 10, 30, 12.817653329393977)
        ]
    )
    return data
```

Then you can also take into account the surface type. Maybe one for each entry with the help of another column or a general value your fixture returns. Remember returning a tuple is possible, you can unpack it inside the test function like:

```python
@pytest.fixture
def foo_bar()
    return "foo", "bar"  # returns a tuple with two strings

def test_foo(foo_bar):
    word1, word2 = foo_bar
    assert word1 == "foo"
```

And if you use a dataframe in its own form, you can unpack it like you would do to a dictionary into keyword args:

```python
@pytest.fixture
def foo_bar():
    return pd.DataFrame(data=[["foo", "bar"], ["echedey", "Luis"]], columns=["a","b"])  # returns a dataframe with two columns

def func_under_test(a, b):
    return a + b

def test_foo(foo_bar):
    result = func_under_test(**foo_bar)  # unpacks like a dictionary into series, IIRC
    assert (result == pd.Series(["foobar", "echedeyLuis"])).all()
```

Let me insist in that this last point is just to share that there are other patterns/tools to make tests, that may make them a bit more readable, and, most importantly, keep data nearby. I only request changes regarding 1. If you need anything else, clarifying or whatever, you know how to DM me.

Reply:
Thank you, @echedey-ls; this was very informative! I tried to implement these suggestions in the tests I wrote.